osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -410,19 +413,24 @@ protected override void Reset(bool storeResults)
         {
             base.Reset(storeResults);
 
-            scoreResultCounts.Clear();
             hitEvents.Clear();
             lastHitObject = null;
 
             if (storeResults)
+            {
                 maximumScoringValues = currentScoringValues;
+                maximumResultCounts = new Dictionary<HitResult, int>(scoreResultCounts);
```

Comment:
Not sure on storing these as separate dictionaries to the `maximumScoringValues` / `currentScoringValues` structs (and whether those should even be structs in the first place). But maybe this is okay for the time being...